domino/domino.py

```diff
@@ -818,6 +819,12 @@ def hardware_tiers_list(self):
         url = self._routes.hardware_tiers_list(self._project_id)
         return self._get(url)
 
+    def get_hardware_tier_id_from_name(self):
```

Comment:
AFAIK, hardware tier names are not required to be unique, so this will be an issue. I think this why overriding by name was probably a mistake in the first place.

It's unfortunate, but I think to address this issue, we really need to change the interface of `job_start()` to take hardware ID from the start. At least we know that no customer is using this as is (because it's totally broken, and no one has reported a bug before now).

Reply:
An alternative to changing the interface would be to accept the name as exists right now, get the ID like you proposed, but throw an error in the event that more than one HWT with the same name is found.

I'm honestly not sure which is the better option.

Reply:
perhaps similar to `on_demand_spark_cluster_properties`, I can document it as deprecated and try the best effort approach and fail if multiple HWT with the same name are found. This way if any customer is using it but not noticing it doesn't work, their code will at least still run.

I can also add a new optional property for HWT id that is recommended to be used instead.

Reply:
@dknupp Does this approach seem ok?

Reply:
@ddl-awroblicky -- ah, I saw this was still marked as a draft, so I didn't know if there were more commits coming. I'll try to have a look later this afternoon or tomorrow.